src/components/account/details/details.js

```diff
@@ -8,151 +8,173 @@ import {
   Card,
   CardHeader,
   CardContent,
-  CardActions,
   Divider,
   Grid,
   Button,
-  TextField
+  LinearProgress,
+  CardActions
 } from "@material-ui/core";
 
-const useStyles = makeStyles(() => ({
-  root: {}
+import { Formik, Field } from "formik";
+
+import { TextField } from "formik-material-ui";
+
+import { UserContext } from "../../../states";
+
+import { createAuthenticatedClient } from "../../../authentication";
+
+import { updateUserById } from "../../../pages/account/api";
+
+const useStyles = makeStyles(theme => ({
+  root: {},
+  submit: {
+    marginTop: "10px"
+  }
 }));
 
+const FormikAccount = ({ classes, user, props }) => (
+  <Formik
+    enableReinitialize={true}
+    initialValues={{ ...user, password: "" }}
+    validate={values => {
+      const errors = {};
+
+      if (!values.email) {
+        errors.email = "Required";
+      } else if (
+        !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)
+      ) {
+        errors.email = "Invalid email address";
```

Comment:
> This can be done very fast using Yup:
> 
> https://github.com/jquense/yup#stringminlimit-number--ref-message-string--function-schema
> 
> https://jaredpalmer.com/formik/docs/guides/validation#validationschema

I agree :+1:
I'll use Yup to validate schemas, it's so much easy than regex. :smiley: